### PR TITLE
docs: Improve ADC api documentation

### PR DESCRIPTION
**Pull request by brianfr-amazon**
Improving ADC api documentation

Why: ADC header needed clarification on proper use of return codes,
and on what data types to expect for some of the api's.

Checklist:
----------

- [ ] I have tested my changes. No regression in existing tests.
- [ ] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: libraries/abstractions/common_io/include/iot_adc.h [hunk 1]
```diff
@@ -146,25 +263,25 @@ int32_t iot_adc_read_sample( IotAdcHandle_t const pxAdc,
...
+                               *!< Shortest sample time maximize conversion speed for lower impedance input
+                               *!< Extending sample time improve sample accuracy for higher impedance input */
+    uint8_t ucAdcResolution;  /*!< ADC channel resolution, value = resolution bits; 12 = 12bit, 13 = 13bit, etc */
 } IotAdcConfig_t;
 
 typedef enum
 {
-    eChStateIdle,
-    eChStateBusy,
+    eChStateIdle,     /*!< ADC is idle. */
```
Comment (vjjunnut) [inline libraries/abstractions/common_io/include/iot_adc.h]: nit - ADC channel is idle. and ADC channel is busy

File: libraries/abstractions/common_io/include/iot_adc.h [hunk 2]
```diff
@@ -33,65 +33,98 @@
...
 IotAdcHandle_t iot_adc_open( int32_t lAdc );
 
 /**
  * @brief Close ADC controller.
- *
  * All pending operation will be cancelled, put ADC module in reset state or low
  * power state if possible. Release all resources claimed during open call
  *
- * @param[in] pxAdc handle to ADC controller
+ * @param[in] pxAdc handle to ADC controller returned from open() call.
```
Comment (lundinc2) [inline libraries/abstractions/common_io/include/iot_adc.h]: Nit: change open() to iot_adc_open()

File: libraries/abstractions/common_io/include/iot_adc.h [hunk 3]
```diff
@@ -33,65 +33,98 @@
 /**
  * @brief Error code returned by ADC driver
  */
-#define IOT_ADC_SUCCESS                   ( 0 )
-#define IOT_ADC_FAILED                    ( 1 )
-#define IOT_ADC_INVALID_VALUE             ( 2 )
-#define IOT_ADC_NOT_OPEN                  ( 3 )
-#define IOT_ADC_FUNCTION_NOT_SUPPORTED    ( 4 )
-#define IOT_ADC_CH_BUSY                   ( 5 )
+#define IOT_ADC_SUCCESS                   ( 0 )     /*!< ADC operation completed succesfully. */
```
Comment (lundinc2) [inline libraries/abstractions/common_io/include/iot_adc.h]: Nit: succesfully needs another s

File: libraries/abstractions/common_io/include/iot_adc.h [hunk 4]
```diff
@@ -195,27 +312,27 @@ typedef struct IotAdcChBuffer_s
...
-    uint8_t ucBufLen;     /* data buffer length, shalll be large enough to hold group sample data */
-    uint16_t usChainMask; /* define which ADC channels are in the chain group.
-                           * e.g. 'x' bit set to 1 means ADC channel x is inclued in the group.
-                           * Client shall manage potential ADC channel use conflict. */
+    uint8_t ucAdcChannel; /*!< logical ADC channel number as input to ioctl. It is recommended
+                           *!< that client uses the first ADC channel number in the group as its
+                           *!< logical number which client can use later to trigger group sampling */
+    void * pvBuffer;      /*!< data buffer used by driver to save converted sample data.
+                           *!< the buffer is allocated by client and passed to driver to use */
+    uint8_t ucBufLen;     /*!< data buffer length, shalll be large enough to hold group sample data */
```
Comment (lundinc2) [inline libraries/abstractions/common_io/include/iot_adc.h]: an extra l in shall

**Pull request merged**